packages/riipen-ui/src/components/Badge.test.jsx

```diff
@@ -0,0 +1,306 @@
+      const component = "abcdefg";
+
+      mount(<Badge component={component} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("content prop", () => {
+    it("badge does not render when content is null and showZero is false", () => {
+      const wrapper = mount(<Badge />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("hidden")
+      ).toEqual(true);
+      expect(wrapper.text()).toEqual("");
+    });
+
+    it("badge renders with content and showZero props", () => {
+      const content = 2;
+      const showZero = true;
+
+      const wrapper = mount(<Badge content={content} showZero={showZero} />);
+
+      expect(Number(wrapper.find("Badge").text())).toEqual(2);
+    });
+  });
+
+  describe("max prop", () => {
+    it("displays max value given max greater than content value", () => {
+      const max = 9;
+      const content = 11;
+
+      const wrapper = mount(<Badge max={max} content={content} />);
+
+      expect(wrapper.text()).toEqual(`${max}+`);
+    });
+
+    it("displays content value given content greater than max value", () => {
+      const max = 9;
+      const content = 8;
+
+      const wrapper = mount(<Badge max={max} content={content} />);
+
+      expect(Number(wrapper.text())).toEqual(content);
+    });
+  });
+
+  describe("overlap prop", () => {
+    it("sets overlap class name with valid custom overlap", () => {
+      const anchor = { horizontal: "left", vertical: "top" };
+      const overlap = "circle";
+
+      const wrapper = mount(<Badge anchor={anchor} overlap={overlap} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("top-left-circle")
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid overlap", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Badge overlap="square" />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("showZero prop", () => {
+    it("badge renders when showZero prop value is true", () => {
+      const showZero = true;
+
+      const wrapper = mount(<Badge showZero={showZero} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("hidden")
+      ).toEqual(false);
+    });
+  });
+
+  describe("size prop", () => {
+    it("sets size class name with a valid size", () => {
+      const size = "small";
+      const variant = "standard";
+
+      const wrapper = mount(<Badge size={size} variant={variant} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(size)
+      ).toEqual(true);
+    });
+
+    it("does not set size class name when invalid variant prop used", () => {
```

Comment:
Instead of "invalid variant" I would be tempted to say "non standard". This makes it more clear to anyone reading the test that invalid actually means non standard.

Reply:
Can I revise it to "non standard variant" to convey that it's the variant prop controlling if size can be used?

Reply:
Love it